src/test/groovy/graphql/nadel/schema/OverallSchemaGeneratorTest.groovy

```diff
@@ -217,4 +218,63 @@ class OverallSchemaGeneratorTest extends Specification {
         schema.getSubscriptionType() == null
 
     }
+
+    def "extend types works"() {
+        when:
+        def schema = TestUtil.schemaFromNdsl("""
+        service S1 {
+            type Query {
+                a: String
+            }
+            extend type Query {
+                b: String
+            }
+            type A {
+                x: String
+            }
+            extend type A {
+                y: String
+            }
+        } 
+        service S2 {
+            type Query {
+                c: String
+            }
+            extend type Query {
+                d: String
+            }
+            extend type A {
+                z: String 
+            }
+        }
+        """)
+        then:
+        schema.getQueryType().fieldDefinitions.collect { it.name } == ['a', 'b', 'c', 'd']
+        (schema.getType("A") as GraphQLObjectType).fieldDefinitions.collect {
+            it.name
+        } == ['x', 'y', 'z']
+
+    }
+
+    def "only extend Query works"() {
+        when:
+        def schema = TestUtil.schemaFromNdsl("""
+        service S1 {
+            extend type Query {
+                a: String
+            }
+            extend type Query {
+                b: String
+            }
+        } 
+        service S2 {
+            extend type Query {
+                c: String
+            }
+        }
+        """)
+        then:
+        schema.getQueryType().fieldDefinitions.collect { it.name } == ['a', 'b', 'c']
+
```

Comment:
You dont have a test for when there is NO query in a service.

I think this should be covered

Reply:
Ignore me!  Yes you do!